src/parser/ad_parser.js

```diff
@@ -22,32 +22,42 @@ export function parseAd(
   emit,
   { allowMultipleAds, followAdditionalWrappers } = {}
 ) {
-  const childNodes = adElement.childNodes;
-
-  for (const adTypeElementKey in childNodes) {
-    const adTypeElement = childNodes[adTypeElementKey];
-
-    if (['Wrapper', 'InLine'].indexOf(adTypeElement.nodeName) === -1) {
-      continue;
-    }
+  const childNodes = Array.from(adElement.childNodes);
+
+  const filteredChildNodes = childNodes
+    .filter((childNode) =>
+      ['inline', 'wrapper'].includes(childNode.nodeName.toLowerCase())
+    )
+    .filter(
+      (adType) =>
+        !(
+          adType.nodeName.toLowerCase() === 'wrapper' &&
+          followAdditionalWrappers === false
+        )
+    );
```

Comment:
Since followAdditionalWrapper can be undefined, i think the filter will be more precise if we do it this way.

const filteredChildNodes = childNodes.filter((childNode) => {
const childNodesToLowerCase = childNode.nodeName.toLowerCase();
return (
childNodesToLowerCase === 'inline' ||
(followAdditionalWrappers !== false &&
childNodesToLowerCase === 'wrapper')
);
});
`